mycpp/gc_mylib.h

```diff
@@ -132,16 +132,42 @@ class Writer : public Obj {
   virtual bool isatty() = 0;
 };
 
-class Buf;  // forward declaration
+class MutableStr;
 
-Str* StrFromBuf(const Buf&);
+class Buf {
+ public:
+  // The initial capacity is big enough for a line
+  Buf(int cap);
+  void Extend(Str* s);
+
+ private:
+  friend class BufWriter;
+  friend Str* StrFromBuf(Buf*);
+  friend Buf* NewBuf(int);
+
+  char* data();
+  char* end();
+  int capacity();
+
+  MutableStr* str_;
+  int len_;
+
+  /*
+  // TODO: move this state into BufWriter
```

Comment:
Why not have just BufWriter and MutableStr, instead of BufWriter, Buf, and MutableStr, as this comment suggests?